libs/mutantschemata/cpp_source/type.hpp

```diff
@@ -0,0 +1,42 @@
+/**
+Copyright: Copyright (c) 2019, Niklas Pettersson. All rights reserved.
+License: MPL-2
+Author: Niklas Pettersson (nikpe353@student.liu.se)
+
+This Source Code Form is subject to the terms of the Mozilla Public License,
+v.2.0. If a copy of the MPL was not distributed with this file, You can obtain
+one at http://mozilla.org/MPL/2.0/.
+
+C++ types for easier communication between C++ and D code.
+*/
+#ifndef TYPE_CPP
+#define TYPE_CPP
+
+#include "cpp_string.hpp"
+
+namespace CppType {
+
+struct SourceLoc {
+    unsigned line;
```

Comment:
Be precise about the size of the variables by using types from `cstdint`. This is e.g. platform dependent on what size this is.